src/main/java/com/scottlogic/filters/DatePostFilter.java

```diff
@@ -13,8 +13,8 @@
     private final OffsetDateTime endDate;
 
     public DatePostFilter(OffsetDateTime startDateInput, OffsetDateTime endDateInput) {
-        startDate = startDateInput;
-        endDate = endDateInput;
+        this.startDate = startDateInput;
+        this.endDate = endDateInput;
```

Comment:
These `this` are generally standard practise but the other side of that is changing the constructor inputs from `startDateInput` to `startDate` as then the `this` has an impact and it can help make the inputs clearer when Intellij picks them up. This is because when Intellij works out what to show you it will show the names of these variables and so having `startDate` and `endDate` is clearer than also having it with the `input` simply because of space and the fact that `input` is redundant information for a constructor as all the values will be `input`s